test/FlashLoan.t.sol

```diff
@@ -0,0 +1,262 @@
+        DAppOperation memory dAppOp = txBuilder.buildDAppOperation(governanceEOA, userOp, solverOps);
+
+        // Frontend signs the dAppOp payload
+        (sig.v, sig.r, sig.s) = vm.sign(governancePK, atlasVerification.getDAppOperationPayload(dAppOp));
+        dAppOp.signature = abi.encodePacked(sig.r, sig.s, sig.v);
+
+        // make the actual atlas call that should revert
+        vm.startPrank(userEOA);
+        vm.expectEmit(true, true, true, true);
+        emit FastLaneErrorsEvents.SolverTxResult(address(solver), solverOneEOA, true, false, 1_048_578);
+        vm.expectRevert();
+        atlas.metacall({ userOp: userOp, solverOps: solverOps, dAppOp: dAppOp });
+        vm.stopPrank();
+
+        // now try it again with a valid solverOp - but dont fully pay back
+        solverOps[0] = txBuilder.buildSolverOperation({
+            userOp: userOp,
+            solverOpData: abi.encodeWithSelector(SimpleSolver.onlyPayBid.selector, 1e18),
+            solverEOA: solverOneEOA,
+            solverContract: address(solver),
+            bidAmount: 1e18,
+            value: 10e18
+        });
+
+        (sig.v, sig.r, sig.s) = vm.sign(solverOnePK, atlasVerification.getSolverPayload(solverOps[0]));
+        solverOps[0].signature = abi.encodePacked(sig.r, sig.s, sig.v);
+
+        dAppOp = txBuilder.buildDAppOperation(governanceEOA, userOp, solverOps);
+        (sig.v, sig.r, sig.s) = vm.sign(governancePK, atlasVerification.getDAppOperationPayload(dAppOp));
+        dAppOp.signature = abi.encodePacked(sig.r, sig.s, sig.v);
+
+        // Call again with partial payback, should still revert
+        vm.startPrank(userEOA);
+        vm.expectEmit(true, true, true, true);
+        emit FastLaneErrorsEvents.SolverTxResult(address(solver), solverOneEOA, true, false, 1_048_578);
+        vm.expectRevert();
+        atlas.metacall({ userOp: userOp, solverOps: solverOps, dAppOp: dAppOp });
+        vm.stopPrank();
+
+        // final try, should be successful with full payback
+        solverOps[0] = txBuilder.buildSolverOperation({
+            userOp: userOp,
+            solverOpData: abi.encodeWithSelector(SimpleSolver.payback.selector, 1e18),
+            solverEOA: solverOneEOA,
+            solverContract: address(solver),
+            bidAmount: 1e18,
+            value: 10e18
+        });
+
+        (sig.v, sig.r, sig.s) = vm.sign(solverOnePK, atlasVerification.getSolverPayload(solverOps[0]));
+        solverOps[0].signature = abi.encodePacked(sig.r, sig.s, sig.v);
+
+        dAppOp = txBuilder.buildDAppOperation(governanceEOA, userOp, solverOps);
+        (sig.v, sig.r, sig.s) = vm.sign(governancePK, atlasVerification.getDAppOperationPayload(dAppOp));
+        dAppOp.signature = abi.encodePacked(sig.r, sig.s, sig.v);
+
+        // Last call - should succeed
+        vm.startPrank(userEOA);
+        atlas.metacall({ userOp: userOp, solverOps: solverOps, dAppOp: dAppOp });
+        vm.stopPrank();
```

Comment:
We should also make some balance checks to ensure the accounting has been done correctly. Atlas balance should be higher of what it was before calling metacall.

Also maybe another test where the bundler sends value along metacall, that should be repaid.